sdks/python/apache_beam/runners/worker/sdk_worker_main.py

```diff
@@ -173,32 +173,47 @@ def create_harness(environment, dry_run=False):
   return fn_log_handler, sdk_harness, sdk_pipeline_options
 
 
+def _start_profiler(gcp_profiler_service_name, gcp_profiler_service_version):
+  try:
+    import googlecloudprofiler
+    if gcp_profiler_service_version:
+      googlecloudprofiler.start(
+          service=gcp_profiler_service_name,
+          service_version=gcp_profiler_service_version,
+          verbose=1)
+      _LOGGER.info('Turning on Google Cloud Profiler.')
+    else:
+      raise RuntimeError('Unable to find the job id from envvar.')
+  except Exception as e:  # pylint: disable=broad-except
+    _LOGGER.warning(
+        'Unable to start google cloud profiler due to error: %s. For how to '
+        'enable Cloud Profiler with Dataflow see '
+        'https://cloud.google.com/dataflow/docs/guides/profiling-a-pipeline.'
+        'For troubleshooting tips with Cloud Profiler see '
+        'https://cloud.google.com/profiler/docs/troubleshooting.' % e)
+
+
+def _get_gcp_profiler_name_if_enabled(sdk_pipeline_options):
```

Comment:
1. Noting that prior version of the code verified whether os.environ["JOB_NAME"] , os.environ["JOB_ID"] variables were defined. Did you intentionally drop these checks?
2. After taking a look `_get_gcp_profiler_name_if_enabled` it looks like we could merge this logic into `GoogleCloudOptions.get_cloud_profiler_service_name()` by adding something like:

```
  def get_cloud_profiler_service_name(self):
      # ...
      # ...
      # existing code
      
      experiments = self.view_as(DebugOptions).experiments or [] 
      ... a few more checks
      
      ...
      return None
```